projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php

```diff
@@ -355,6 +356,38 @@ function wpcom_get_domain_upsell_badge_text() {
 	return wpcom_is_checklist_task_complete( 'domain_upsell' ) ? '' : __( 'Upgrade plan', 'jetpack-mu-wpcom' );
 }
 
+/**
+ * Determines whether or not domain upsell task should be visible.
+ *
+ * @return bool True if user is on a free plan with no custom domain name.
+ */
+function wpcom_is_domain_upsell_visible() {
+	if ( ! function_exists( 'wpcom_get_site_purchases' ) ) {
+		return false;
+	}
+
+	$site_purchases = wpcom_get_site_purchases();
+
+	$bundle_purchases = array_filter(
+		$site_purchases,
+		function ( $site_purchase ) {
+			return $site_purchase->product_type === 'bundle';
+		}
+	);
+
+	$is_free_plan = empty( $bundle_purchases );
```

Comment:
Non-blocking: It may be worth using this as an early return condition instead.
```suggestion
	// If we have a paid plan, we don't want to show the domain upsell task.
	if ( ! empty( $bundle_purchases ) ) {
		return false;
	}
```